decisions/log/0026-async-request-by-default.md

```diff
@@ -0,0 +1,180 @@
+# 26. Switch to async request/response as mandatory (default) behaviour.
+
+Date: 2023-12-05 (updated)
+
+## Status
+
+Proposed
+
+## Context
+
+The current version of the specification[^1] favors synchronous communication over asynchronous communication. As, especially in the early stages of the network, most requests will target not yet existing PF values, the existing approach using synchronous `ListFootprints` calls will not work. PCF data is often not available and is calculated after the request for a specific product arrives at the data provider (Supplier). 
+
+Furthermore managing the access rights upfront for `ListFootprints` requests are complex for large enterprises with different Business Units and many products. Routing requests on specific products is easier than providing PCF data and managing access rights upfront. 
+
+The seconds synchronous request is `GetFootprint` which uses PF IDs to select the requested PCF. But _PF IDs_ are a detail which is usually unknown to the data requester (customer). Data requesters usually only know _Material Identifiers_ (e.g. catalog number, part number).
+Product specific requests would in addition enable data providers (suppliers) to prioritize the calculation of PCF when these are not available yet.
+
+The specification[^1] already adresses all this by supporting `PF Request Events and Responses`. But this feature is _optional_ at the moment whereas synchronous communication mechanisms (as `ListFootprints` or `GetFootprint`) are defined as _mandatory_. This complicates or even block integration of existing PCF exchange apps as those apps usually work in an asynchrounous and productId focused way.
```

Comment:
To 1)

Core of the problem is, that you have to consider that PCF is NOT available at request time. As _filters_ are an _optional_ feature neither the data owner nor the consumer can rely on it to be supported/used. Means, a data consumer has to deal with "generic" ListFootprint calls without any correlation to e.g. a product. 

As a data provider that works product centric and calculates PCF values on demand (the polls Beth provided confirmed that both approaches are quite common), i would not be able to send a valid answer (I will only have HTTP-200 - "Ok, here are the results" and HTTP-202 - "Accepted, I'll deal with that") without a bigger out-of-bound communication with my customer up front. So a "pure" API solution is not possible for me in such a scenario.


To 2)

I would expect that once I have a correlation between the PCF value and a consumer/customer - either made out of bound up front or in a pure API way using the async call (second approach preferred ;) ) - I can provide this PF also synchronously to this specific customer via ListFootprints. 

I selected CAN instead of MUST here, as this might be a data security relevant feature within an app. So there may be an additional step to be performed in an exchange app (e.g. "mark as directly shareable"). But my expectation would be that it can and will be made available also synchronously.

Nevertheless MAY the consumer also ask asynchronously again and the provider will have to send and async answer back again in this case.